### DISCUSSION TITLE
你好，可以出一期讲解多线程、异步、并行相关的视频吗？非常感谢~

### DISCUSSION THREAD

**Discussion by tawtg** [BYJRK/DotNet-Discussions#51] [C# 技术交流]
非常喜欢你的视频，直击痛点，拜托了...

**Comment by BYJRK** [comment]
我需要一些时间准备，因为这个话题很大，细节和技巧很多而且比较复杂，准备不充分难免会被喷😔

**Reply by yang0biao** [reply]
多线程太难了😔，我开了一个单独线程进行一些计算，却不会控制它暂停、继续、结束

**Reply by Dynesshely** [reply]
> 多线程太难了😔，我开了一个单独线程进行一些计算，却不会控制它暂停、继续、结束

考虑实现类似 `CancellationToken` 的类, 在线程中通过查询该类下成员的值来控制线程的运行状态

**Reply by SamaHddd** [reply]
最近在用Backgroundworker这个类，感觉还行

**Comment by WWMMFFLLRR** [comment]
大佬，最近遇到的问题类似，后台每收到一包大数据，就开一个线程触发事件发出去，UI线程订阅事件，FFT等数学处理后，更新UI的image控件，用了Dispatcher还是比较卡，大概50ms一包数据。有没有好的解决方案呢？
还有个问题是C#中深复制一个对象，只能通过序列化来实现吗？

**Reply by KiyanYang** [reply]
深复制的方法很多，比如 https://www.codeproject.com/Articles/1111658/Fast-Deep-Copy-by-Expression-Trees-C-Sharp

**Reply by TruePluto** [reply]
> 大佬，最近遇到的问题类似，后台每收到一包大数据，就开一个线程触发事件发出去，UI线程订阅事件，FFT等数学处理后，更新UI的image控件，用了Dispatcher还是比较卡，大概50ms一包数据。有没有好的解决方案呢？
> 还有个问题是C#中深复制一个对象，只能通过序列化来实现吗？

总能从这个论坛学到新东西 深拷贝 的最安全办法 序列化 谢谢

**Reply by eijjxx** [reply]
更新image的确会比较消耗资源
你这个50ms一包的数据是20帧
是否fft之类的工作在UI线程里做了？如果ui线程里只更新图片的话问题应该不大啊
是不是你的图片特别大？如果图片大的话就考虑用工作线程压缩一下再显示。

还有个邪道方法，用VisualTarget实现多UI线程，这玩意创建的控件是不能直接交互的，如果只是显示图片应该问题不大

**Comment by xianxianyise** [comment]
期待您的线程方面的授课
